cucumber/src/main/java/eu/drus/jpa/unit/cucumber/EqualsInterceptor.java

```diff
@@ -1,35 +1,22 @@
 package eu.drus.jpa.unit.cucumber;
 
-import java.lang.reflect.Method;
-
-import net.sf.cglib.proxy.Callback;
-import net.sf.cglib.proxy.Factory;
-import net.sf.cglib.proxy.MethodInterceptor;
-import net.sf.cglib.proxy.MethodProxy;
-
-public class EqualsInterceptor implements MethodInterceptor {
-
-    private final Object target;
-
-    public EqualsInterceptor(final Object target) {
-        this.target = target;
-    }
-
-    @Override
-    public Object intercept(final Object proxy, final Method method, final Object[] args, final MethodProxy methodProxy) throws Throwable {
-        final Object other = args[0];
-
-        if (proxy == other) {
+import net.bytebuddy.implementation.bind.annotation.Argument;
+import net.bytebuddy.implementation.bind.annotation.FieldValue;
+import net.bytebuddy.implementation.bind.annotation.RuntimeType;
+import net.bytebuddy.implementation.bind.annotation.This;
+
+public final class EqualsInterceptor {
+	
+	private EqualsInterceptor() {}
+
+	@RuntimeType
+	public static Object intercept(@This Object thiz, @Argument(0) Object other, @FieldValue("bean") Object bean) throws Exception {			
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)